src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+                "generation_for_own_use",
+            }
+        )
+        return df.loc[~(generation_mask & is_aggregated_mask)]
+
+    def _collect_totals(df: pd.DataFrame) -> pd.DataFrame:
+        """Various total columns are named differently - they don't have to be."""
+        return df.assign(
+            dimension=df.dimension.str.replace(r"^total.*$", "total", regex=True)
+        )
+
+    def _decumulate(df: pd.DataFrame, decumulate_columns: list[str]) -> pd.DataFrame:
+        """De-accumulate the cumulative columns so we get yearly values.
+
+        Cumulative values are less useful because they require the user to know
+        what year the accumulation starts.
+        """
+        new_columns = {}
+        for decumulate_column in decumulate_columns:
+            cumulative_colname = f"cumulative_{decumulate_column}"
+            column = df[cumulative_colname]
+            new_columns[decumulate_column] = column.diff().fillna(column)
+        return df.assign(**new_columns).drop(
+            columns=[f"cumulative_{c}" for c in decumulate_columns]
+        )
+
+    sanitized = (
+        filter_enrich_sanitize(
+            raw_df=raw_eiaaeo__electric_power_projections_regional,
+            relevant_series_names=(
+                "Electricity : Electric Power Sector : Capacity",
+                "Electricity : Electric Power Sector : Generation",
+                "Electricity : Electric Power Sector : Cumulative Retirements",
+                "Electricity : Electric Power Sector : Cumulative Planned Additions",
+                "Electricity : Electric Power Sector : Cumulative Unplanned Additions",
+                "Electricity : Electric Power Sector : Cumulative Total Additions",
+            ),
+        )
+        .pipe(_drop_aggregated_generation_data)
+        .pipe(_collect_totals)
+    )
+
+    assert set(sanitized.topic.unique()) == {"electricity"}
+    assert set(sanitized.subtopic.unique()) == {"electric_power_sector"}
+
+    trimmed = sanitized.drop(
+        columns=[
+            "topic",
+            "subtopic",
+            "units",
+        ]
+    )
+
+    unstacked = unstack(
+        df=trimmed,
+        eventual_pk=[
+            "report_year",
+            "model_case_eiaaeo",
+            "region",
+            "dimension",
+            "projection_year",
+        ],
+    )
+
+    # check that the totals all add up *before* decumulating, because
+    # decumulating introduces some small amount more floating point error.
+    ratio_close_additions_to_total, ratio_close_reported_calculated = (
+        check_totals_add_up(unstacked)
+    )
+    assert (
+        0.999 < ratio_close_additions_to_total <= 1.0
+    ), f"planned + unplanned vs. total: {ratio_close_additions_to_total}"
+    assert (
+        0.999 < ratio_close_reported_calculated <= 1.0
+    ), f"reported vs. calculated totals: {ratio_close_reported_calculated}"
+    # Now that the totals add up, we can drop the total_additions col
```

Comment:
light nit but i think it would be cleaner to put these asserts into `check_totals_add_up` unless you really need these two totals for later.

Reply:
It was nice to return the totals in development, and IMO "figure out how close the totals are" could be a pretty generic function - if we want to call it in different places we'd need to pass in different thresholds, error messages, etc. So I figured I'd keep the *called* function simpler and put the error handling stuff in the caller. This will end up getting refactored a bit in #3597 since that has some total-summing too - let's discuss best structure there once this is merged!